### PR TITLE
Update engines.

### PR DESCRIPTION
**Pull request by mvaled**
I'm trying to add this in a new project, but fails because my node version is 19.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,11 +30,11 @@
     "dist"
   ],
   "packageManagers": {
-    "pnpm": "7.26.1"
+    "pnpm": "8.2.0"
```
Comment (jihchi) [inline package.json:33]: Please revert this change.

File: package.json [hunk 2]
```diff
@@ -30,11 +30,11 @@
...
   },
   "engines": {
-    "node": "^14 || ^16 || ^18",
-    "pnpm": "^7.26.1"
+    "node": "^18 || ^19",
```
Comment (jihchi) [inline package.json:36]:
```suggestion
    "node": "^14.18.0 || >=16.0.0",
```

It is better to align with [Vite](https://github.com/vitejs/vite/blob/ad21ec35de8b6745e30f6982a0e29a1cf97d55c8/package.json#L6).

File: package.json [hunk 3]
```diff
@@ -30,11 +30,11 @@
...
+    "pnpm": "8.2.0"
```
Comment (jihchi) [inline package.json:37]: Please revert this change.

**Review by jihchi — CHANGES_REQUESTED** [review]
I've addressed the version of pnpm directly, we can exclude it in this PR.

File: package.json (hunk 2 above)
Comment (mvaled) [reply package.json]: Done.

File: package.json (hunk 3 above)
Comment (mvaled) [reply package.json]: Done.

File: package.json (hunk 1 above)
Comment (mvaled) [reply package.json]: Done.

**Review by jihchi — APPROVED** [review]
Awesome! Thank you for the contribution.

**Review by jihchi — APPROVED** [review]

**Comment by jihchi** [comment]
@mvaled Yes, here is the new release: `@jihchi/vite-plugin-rescript@5.2.0`
